packages/fxa-auth-server/lib/routes/subscriptions.js

```diff
@@ -259,21 +255,14 @@ module.exports = (log, db, config, customs, push, mailer, subhub, profile) => {
 
         await subhub.cancelSubscription(uid, subscriptionId);
 
-        try {
-          await db.cancelAccountSubscription(uid, subscriptionId, Date.now());
-        } catch (err) {
-          if (err.statusCode === 404 && err.errno === 116) {
-            throw error.subscriptionAlreadyCancelled();
-          }
-        }
+        await updateSubscriptionsFromSubhub({ db, subhub, profile, uid });
```

Comment:
Hmm, what might be better here is to run `subhub.cancelSubscription` and catch any errors from that. Then, call `updateSubscriptionsFromSubhub` no matter what and ignore any errors it might throw. Finally, re-throw any error caught from `subhub,cancelSubscription`

> I think the fix would be to try/catch the cancelSubscription call, and if it fails because the subscription is already canceled, then just swallow the error and let updateSubscriptionsFromSubhub run. I think?

I'm kind of thinking the opposite, because the error from subhub would be the important one to report (and more likely). This new `updateSubscriptionsFromSubhub` method should be run often and in many spots in order to kind of self-heal any inconsistencies when possible.